src/Token/Token.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {forwardRef, KeyboardEventHandler, MouseEventHandler} from 'react'
+import styled from 'styled-components'
+import {get} from '../constants'
+import TokenBase, {isTokenInteractive, TokenBaseProps} from './TokenBase'
+import RemoveTokenButton from './_RemoveTokenButton'
+
+export interface TokenProps extends TokenBaseProps {
+  /**
+   * A function that renders a component before the token text
+   */
+  leadingVisual?: React.FunctionComponent<any>
```

Comment:
Does this need to a be a FunctionComponent or could it be `React.ComponentType`?

Reply:
I think it could be `React.ComponentType`